packages/devtools_app/lib/src/screens/inspector/inspector_tree_controller.dart

```diff
@@ -773,6 +775,12 @@ class _InspectorTreeState extends State<InspectorTree>
     }
     _focusNode = FocusNode(debugLabel: 'inspector-tree');
     autoDisposeFocusNode(_focusNode);
+    addConditionalAutoDisposeListener(
```

Comment:
Instead of using auto_dispose_mixin, you can just listen to _doneBindingToController and dispose the controller in the listener.
Will it be more readable and take less code then?

Reply:
I chatted with Kenzie about this a bit and we decided to pivot the approach a bit so that we can still make things a bit easier for the caller. i've tweaked the helper so that it will make it much easier for us to run once now :D